### PR TITLE
MAT-239 – expand Gift Aid claim result handling

### PR DESCRIPTION
**Pull request by NoelLH**

### CODE REVIEW

File: src/Application/Messenger/Handler/GiftAidResultHandler.php [hunk 1]
```diff
@@ -20,14 +20,28 @@ public function __construct(
...
-            'Donation ID %s Gift Aid claim failure reported by ClaimBot',
+            'Donation ID %s Gift Aid claim result returned by ClaimBot',
             $donationMessage->id,
         ));
 
         /** @var Donation $donation */
         $donation = $this->donationRepository->findOneBy(['uuid' => $donationMessage->id]);
 
-        $donation->setTbgGiftAidRequestFailedAt(new \DateTime());
+        if ($donationMessage->responseSuccess === false) {
```
Comment (alihejazi) [inline src/Application/Messenger/Handler/GiftAidResultHandler.php:30]: `if (!$donationMessage->responseSuccess)`

File: src/Application/Messenger/Handler/GiftAidResultHandler.php [hunk 2]
```diff
@@ -20,14 +20,28 @@ public function __construct(
...
+            $donation->setTbgGiftAidRequestFailedAt(new \DateTime());
+        }
+
+        if ($donationMessage->responseSuccess === true) {
```
Comment (alihejazi) [inline src/Application/Messenger/Handler/GiftAidResultHandler.php:34]: `=== true` is redundant

**Review by alihejazi — APPROVED** [review]

File: src/Application/Messenger/Handler/GiftAidResultHandler.php (hunk 1 above)
Comment (NoelLH) [reply src/Application/Messenger/Handler/GiftAidResultHandler.php:30]: @alihejazi this is not the same. We need to explicitly ensure that it is `false`, not `null`.

File: src/Application/Messenger/Handler/GiftAidResultHandler.php (hunk 2 above)
Comment (NoelLH) [reply src/Application/Messenger/Handler/GiftAidResultHandler.php:34]: I think it is still clearer when we're dealing with a nullable type and one that's so fundamental.

**Pull request merged**